server/src/main/java/org/opensearch/common/time/CustomDateTimeFormatter.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.time;
+
+import java.text.Format;
+import java.text.ParsePosition;
+import java.time.ZoneId;
+import java.time.chrono.Chronology;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.TemporalAccessor;
+import java.time.temporal.TemporalQuery;
+import java.util.Locale;
+
+/**
+* Wrapper class for DateTimeFormatter{@link java.time.format.DateTimeFormatter}
+* to allow for custom implementations for datetime parsing/formatting
+ */
+public class CustomDateTimeFormatter {
```

Comment:
Not fully understand why do we need this class if it delegates everything to `DateTimeFormatter` instance?

Reply:
I think the name is confusing a bit, may be we could just use `org.opensearch.common.time.DateTimeFormatter` (or just `org.opensearch.common.time.Formatter`)?

Reply:
using `org.opensearch.common.time.DateTimeFormatter` is creating class resolution issues where its being used with `java.time.format.DateTimeFormatter`. We've to then mention fully qualified class name everywhere `java.time.format.DateTimeFormatter` is being used. Let me know if `org.opensearch.common.time.OpenSearchDateTimeFormatter` looks good?